docs/_posts/2021-05-19-employing-right-configurations-for-hudi-cleaner.md

```diff
@@ -0,0 +1,77 @@
+ - **File slice**: A file slice consists of the base file and any log files consisting of the delta, in case of MERGE_ON_READ table type.
+ - **Hudi File Group**: Any file group in Hudi is uniquely identified by the partition path and the  file id that the files in this group have as part of their name. A file group consists of all the file slices in a particular partition path. Also any partition path can have multiple file groups.
+
+### Cleaning Policies
+
+Hudi cleaner currently supports below cleaning policies:
+
+ - **KEEP_LATEST_COMMITS**: This is the default policy. This is a temporal cleaning policy that ensures the effect of having lookback into all the changes that happened in the last X commits. Suppose a writer ingesting data  into a Hudi dataset every 30 minutes and the longest running query can take 5 hours to finish, then the user should retain atleast the last 10 commits. With such a configuration, we ensure that the oldest version of a file is kept on disk for at least 5 hours, thereby preventing the longest running query from failing at any point in time. Incremental cleaning is also possible using this policy.
+ - **KEEP_LATEST_FILE_VERSIONS**: This is a static numeric policy that has the effect of keeping N number of file versions irrespective of time. This policy is use-ful when it is known how many MAX versions of the file does one want to keep at any given time. To achieve the same behaviour as before of preventing long running queries from failing, one should do their calculations based on data patterns. Alternatively, this policy is also useful if a user just wants to maintain 1 latest version of the file.
+
+### Examples
+
+Suppose a user uses the below configs for cleaning:
+
+```java
+hoodie.cleaner.policy=KEEP_LATEST_COMMITS
+hoodie.cleaner.commits.retained=10
+```
+
+Cleaner selects the versions of files to be cleaned by taking care of the following:
+
+ - Latest version of a file should not be cleaned.
+ - The commit times of the last 10 (configured) + 1 commits are determined. One extra commit is included because the time window for retaining commits is essentially equal to the longest query run time. So if the longest query takes 5 hours to finish, and ingestion happens every 30 minutes, you need to retain last 10 commits since 10*30 = 300 (5 hours). At this point of time, the longest query can still be using files written in 11th commit in reverse order.  Now for any file group, only those file slices are scheduled for cleaning which are not savepointed (another Hudi table service) and whose commit time is less than the 11th commit in reverse order.
+
+Suppose a user uses the below configs for cleaning:
+
+```java
+hoodie.cleaner.policy=KEEP_LATEST_FILE_VERSIONS
+hoodie.cleaner.fileversions.retained=2
+```
+
+Cleaner does the following:
+
+ - For any file group, last 2 versions (including any for pending compaction) of file slices are kept and the rest are scheduled for cleaning.
```

Comment:
I feel it would be nice if we can give more details here. Even for me it took me a while to understand the details when I first bumped into something regarding cleaning policy.
May be, show an example where there are 5 commits, where in each commit different file groups are touched. And show a timeline of cleaning events on every commit. If not, first time readers might wonder how is this different from previous policy. main crux here is that not every file group will be touched in every commit. 
Also, you could considering doing the same w/ previous policy too since some file groups might have more file versions while others might have less.

Reply:
Thank you for pointing this out. Adding visual examples here.